include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -631,28 +632,37 @@ struct __get_sycl_range
         });
     }
 
-  private:
     //implementation of operator()(_Iter __first, _Iter __last) for the host iterator types
-    template <typename _Iter, typename _GetBufferFunc>
+    template <sycl::access::mode _LocalAccMode, typename _Iter, typename _GetBufferFunc>
     auto
     __process_host_iter_impl(_Iter __first, _Iter __last, _GetBufferFunc __get_buf)
     {
-        static_assert(!oneapi::dpl::__internal::is_const_iterator<_Iter>::value || AccMode == sycl::access::mode::read,
+        static_assert(!oneapi::dpl::__internal::is_const_iterator<_Iter>::value ||
+                          _LocalAccMode == sycl::access::mode::read,
                       "Should be non-const iterator for a modifying algorithm.");
 
         assert(__first < __last);
 
         auto __buf = __get_buf();
-        __buf.set_write_back(__is_copy_back);
+        __buf.set_write_back(__is_copy_back_v<_LocalAccMode>);
 
         // We have to extend sycl buffer lifetime by sync reasons in case of host iterators. SYCL runtime has sync
         // in buffer destruction and a sycl view instance keeps just placeholder accessor, not a buffer.
         using BufferType = oneapi::dpl::__internal::__lifetime_keeper<decltype(__buf)>;
         m_buffers.push_back(::std::make_unique<BufferType>(__buf));
 
         using _T = val_t<_Iter>;
-        return __range_holder<oneapi::dpl::__ranges::all_view<_T, AccMode>>{
-            oneapi::dpl::__ranges::all_view<_T, AccMode>(__buf)};
+        return __range_holder<oneapi::dpl::__ranges::all_view<_T, _LocalAccMode>>{
+            oneapi::dpl::__ranges::all_view<_T, _LocalAccMode>(__buf)};
+    }
+
+  public:
+    template <typename... _ArgType>
+    auto
+    operator()(_ArgType... __arg)
```

Comment:
A minor comment. Usually we use a plural form for "argument pack"  - `"operator()(_ArgTypes... __args)"`

Reply:
Fixed.